src/window/windows/castlewindow_winapi.inc

```diff
@@ -719,6 +728,34 @@ begin
           DoMouseWheel(TWinParam(WParm).HiSmallInt / 120, uMsg = WM_MOUSEWHEEL);
         Exit;
       end;
+
+    { Event received when something in Windows configuration had been changed,
+      most often this means connecting/disconnecting plug-n-play devices.
+      Here we use it to detect DBT_DEVICEARRIVAL and DBT_DEVICEREMOVECOMPLETE events
+      which correspond to adding or removing devices and suggest those are
+      caused by connecting/disconnecting a Joystick from the system. }
+    WM_DEVICECHANGE:
+      begin
+        case WParm of
+          { A device had been sucessfully added to the system }
+          DBT_DEVICEARRIVAL:
+            begin
+              WriteLnLog('Received a WM_DEVICECHANGE event with DBT_DEVICEARRIVAL');
+              //it might be a joystick, therefore
+              Joysticks.InternalConnected;
+            end;
+          { A device had been succesfully removed from the system }
+          DBT_DEVICEREMOVECOMPLETE:
+            begin
+              WriteLnLog('Received a WM_DEVICECHANGE event with DBT_DEVICEREMOVECOMPLETE');
+              //it might be a joystick, therefore
+              Joysticks.InternalDisconnected;
+            end;
+          { Something unspecified changed in the system devices configuration }
+          //DBT_DEVNODES_CHANGED: WriteLnLog('Received a WM_DEVICECHANGE event with DBT_DEVNODES_CHANGED'); // Ignore these events for now
```

Comment:
To actually ignore other events, we should not do `Exit(0)` in this case.

`WindowProc` at the end calls `DefWindowProc` (and other stuff) if you don't exit early.

I don't know what is the best course of action, but either

1. We should do `Exit(0)` only for 2 handled events - DBT_DEVICEARRIVAL, DBT_DEVICEREMOVECOMPLETE (seems more natural)
2. Or remove the comment "Ignore these events for now" in the line above. If we do `Exit(0)`, then we don't ignore them, 

https://docs.microsoft.com/en-us/windows/win32/devio/wm-devicechange docs say about the return value
"""Return TRUE to grant the request.
Return BROADCAST_QUERY_DENY to deny the request.
"""

Which suggests that the proper solution is

3. Always do `Exit(1)`. I don't know why SDL does `Exit(0)`.

Reply:
Hmmm... I'm not sure either. I'm not even sure if we actually need to `Exit` at all. So, let's go with "3".